nf_core/modules/modules_json.py

```diff
@@ -1161,7 +1167,14 @@ def recreate_dependencies(self, repo, org, subworkflow):
         dep_mods, dep_subwfs = get_components_to_install(sw_path)
 
         for dep_mod in dep_mods:
-            installed_by = self.modules_json["repos"][repo]["modules"][org][dep_mod]["installed_by"]
+            try:
+                installed_by = self.modules_json["repos"][repo]["modules"][org][dep_mod]["installed_by"]
+            except KeyError:
+                log.warning(
+                    f"Module '{dep_mod}' is a dependency of '{subworkflow}'\n"
+                    f"You can install it with the command 'nf-core modules install {dep_mod}'"
+                )
+                continue
```

Comment:
can't we handle it more gracefully on our side and just reinstall?

Reply:
Actually, I'm not sure. We will have an import error if we try to install with the install command. 
I saw this error before fixing the `self.pipeline_components` issue, in theory we shouldn't see this in normal conditions. Also, this function runs after we fix all unsynced components in `modules.json`

Reply:
Can we remove it then again?